File: apple.js
class Apple {
    constructor(snake) {
        this.snake = snake;
        this.start();
    }
    
    start() {
        do {
            this.x = Math.floor(27 * Math.random()) * 20;
            this.y = Math.floor(27 * Math.random()) * 20;
        } while (this.eaten());
        this.square = new Square(this.x, this.y, 255, 0, 0);
    }

    new() {
        do {
            this.x = Math.floor(27 * Math.random()) * 20;
            this.y = Math.floor(27 * Math.random()) * 20;
        } while (this.eaten());
        this.square.x = this.x;
        this.square.y = this.y;
    }

    eaten() {
        return this.x == this.snake.squares[0].x && this.y == this.snake.squares[0].y;
    }
}

File: src/components/Apple.java
package components;

import java.util.ArrayList;

import util.Constants;
import util.Square;

public class Apple extends Square {

    public Apple(ArrayList<Square> snake) {
        this.generate(snake);
    }

    public void generate(ArrayList<Square> snake) {
        boolean eaten = false;
        int x, y;
        do {
            x = (int) Math.floor(Constants.WINDOW_SIZE / Constants.SNAKE_SIZE * Math.random()) * Constants.SNAKE_SIZE;
            y = (int) Math.floor(Constants.WINDOW_SIZE / Constants.SNAKE_SIZE * Math.random()) * Constants.SNAKE_SIZE;
            for(Square square : snake) {
                if(x == square.getX() && y == square.getY()) {
                    eaten = true;
                    break;
                }
            }
        } while(eaten);

        this.setX(x);
        this.setY(y);
    }

   
}
